### PR TITLE
Added to_regressor and to_classifier functions

### PR DESCRIPTION
**Pull request by ilya-pchelintsev**
I hereby agree to the terms of the CLA available at: [https://yandex.ru/legal/cla/?lang=en](https://yandex.ru/legal/cla/?lang=en).

### CODE REVIEW

File: catboost/python-package/catboost/core.py [hunk 1]
```diff
@@ -5074,3 +5074,31 @@ def _plot_feature_statistics(statistics, feature, max_cat_features_on_plot):
     else:
         fig = _build_binarized_feature_statistics_fig(statistics, feature)
         return [fig]
+
+
+def to_regressor(model):
+    regressor = CatBoostRegressor()
```
Comment (LyzhinIvan) [inline catboost/python-package/catboost/core.py]: Let's add here a type check for model (CatBoost or CatBoostRegressor). And also let's return model itself immediately if it is CatBoostRegressor already.

Comment (LyzhinIvan) [inline catboost/python-package/catboost/core.py]: Let's avoid `__init__` call here by this way: `regressor = CatBoostRegressor.__new__(CatBoostRegressor)`

File: catboost/python-package/catboost/core.py [hunk 2]
```diff
@@ -5074,3 +5074,31 @@ def _plot_feature_statistics(statistics, feature, max_cat_features_on_plot):
...
+
+    params = deepcopy(model._init_params)
+    _process_synonyms(params)
+    if 'loss_function' in params:
+        regressor._check_is_regressor_loss(params['loss_function'])
+
+    for attr in ['_object', '_init_params', '_random_seed', '_learning_rate', '_tree_count']:
```
Comment (LyzhinIvan) [inline catboost/python-package/catboost/core.py]: This list of attributes is not complete. For example, we loss feature_importance_, _prediction_values_change and _loss_value_change while converting. Maybe, there are some other lost attributes. Check it carefully, please.

File: catboost/python-package/catboost/core.py [hunk 3]
```diff
@@ -5074,3 +5074,31 @@ def _plot_feature_statistics(statistics, feature, max_cat_features_on_plot):
...
+            setattr(regressor, attr, getattr(model, attr))
+    return regressor
+
+
+def to_classifier(model):
+    classifier = CatBoostClassifier()
+
+    params = deepcopy(model._init_params)
+    _process_synonyms(params)
+    if 'loss_function' in params:
```
Comment (LyzhinIvan) [inline catboost/python-package/catboost/core.py]: If we fit model by CLI or base CatBoost class, we may not to specify `loss_function` and it will be set to `RMSE` by default. So `_init_params` will not contain loss_function, but `get_all_params()` will return it.

File: catboost/python-package/ut/medium/test.py [hunk 4]
```diff
@@ -6875,3 +6877,55 @@ def test_langevin_with_empty_leafs():
...
+        models.append(CatBoostClassifier(iterations=2, learning_rate=learning_rate))
+        models[-1].fit(train_pool, eval_set=test_pool)
+
+    merged_model = sum_models(models)
+    prediction = merged_model.predict(test_pool, prediction_type='Probability')
+
+    merged_model = to_classifier(merged_model)
+
+    assert isinstance(merged_model, CatBoostClassifier)
+    assert _check_data(merged_model.predict(test_pool, prediction_type='Probability'), prediction)
```
Comment (LyzhinIvan) [inline catboost/python-package/ut/medium/test.py]: `CatBoostClassifier` has special method `predict_proba`, it will be better to use it here.

File: catboost/python-package/ut/medium/test.py [hunk 5]
```diff
@@ -6875,3 +6877,55 @@ def test_langevin_with_empty_leafs():
...
+def test_to_regressor():
+    train_pool = Pool(TRAIN_FILE, column_description=CD_FILE)
+    test_pool = Pool(TEST_FILE, column_description=CD_FILE)
+
+    models = []
+    for learning_rate in [.03, .05]:
+        models.append(CatBoostRegressor(iterations=2, learning_rate=learning_rate))
+        models[-1].fit(train_pool, eval_set=test_pool)
+
+    merged_model = sum_models(models)
```
Comment (LyzhinIvan) [inline catboost/python-package/ut/medium/test.py]: Let's add explicit assert that merged_model is an object of base type (`CatBoost`).

File: catboost/python-package/catboost/core.py [hunk 6]
```diff
@@ -5077,28 +5077,40 @@ def _plot_feature_statistics(statistics, feature, max_cat_features_on_plot):
...
+    regressor = CatBoostRegressor.__new__(CatBoostRegressor)
 
+    # TODO(ilyzhin) change on get_all_params after MLTOOLS-4758
     params = deepcopy(model._init_params)
     _process_synonyms(params)
     if 'loss_function' in params:
         regressor._check_is_regressor_loss(params['loss_function'])
 
-    for attr in ['_object', '_init_params', '_random_seed', '_learning_rate', '_tree_count']:
+    for attr in ['_object', '_init_params', '_random_seed', '_learning_rate', '_tree_count', '_loss_value_change', '_prediction_values_change']:
```
Comment (annaveronika) [inline catboost/python-package/catboost/core.py]: I think you need to enumerate through all attributes instead of mentioning them here by name, otherwise this code will break when we add any new field. And there is no test that checks that.

**Comment by annaveronika** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@annaveronika, internal review request created: [1151557](https://a.yandex-team.ru/review/1151557)

**Comment by arcadia-devtools** [comment]
Check [1151557](https://a.yandex-team.ru/review/1151557) passed
